### ISSUE TITLE
vLLM backend broken in v2.8.0

### ISSUE THREAD

**Issue by golgeek** [mudler/LocalAI#1710]
**LocalAI version:**
v2.8.0-cublas-cuda12

**Environment, CPU architecture, OS, and Version:**
Ubuntu 22.04, amd64, Docker

**Describe the bug**
Python stacktrace when curling a vllm backend in v2.8.0, while the same configuration works in v2.7.0.

```sh
curl http://localhost:8080/v1/completions -H "Content-Type: application/json" -d '{
  "model": "vllm",
  "prompt": "Hello, my name is",
  "temperature": 0.1, "top_p": 0.1
}'
{"error":{"code":500,"message":"grpc service not ready","type":""}}
```

The following backend error happens when loading the model (see full logs later in the post):
```
4:47PM DBG GRPC(facebook/opt-125m-127.0.0.1:38441): stderr ImportError: /opt/conda/envs/transformers/lib/python3.11/site-packages/vllm/_C.cpython-311-x86_64-linux-gnu.so: undefined symbol: _ZN2at4_ops15to_dtype_layout4callERKNS_6TensorEN3c108optionalINS5_10ScalarTypeEEENS6_INS5_6LayoutEEENS6_INS5_6DeviceEEENS6_IbEEbbNS6_INS5_12MemoryFormatEEE
```

**To Reproduce**
Declare a vllm model as specified [in the documentation](https://localai.io/features/text-generation/#vllm):
```yaml
name: vllm
backend: vllm
parameters:
    model: "facebook/opt-125m"
```

Curl LocalAI v2.8.0 with the documentation example command:
```sh
curl http://localhost:8080/v1/completions -H "Content-Type: application/json" -d '{
  "model": "vllm",
  "prompt": "Hello, my name is",
  "temperature": 0.1, "top_p": 0.1
}'
{"error":{"code":500,"message":"grpc service not ready","type":""}}
```

**Expected behavior**
Working model load as with versions prior to v2.8.0; example with v2.7.0:
```sh
curl http://localhost:8080/v1/completions -H "Content-Type: application/json" -d '{
  "model": "vllm",
  "prompt": "Hello, my name is",
  "temperature": 0.1, "top_p": 0.1
}'
{"created":1707929418,"object":"text_completion","id":"273ff23c-6ce0-4fc6-8d89-6bb62e499f73","model":"vllm","choices":[{"index":0,"finish_reason":"stop","text":" J.C. and I am a student at the University of California, Berkeley. I am a graduate of the University of California, Berkeley, and a graduate of the University of California, Berkeley. I am a graduate of the University of California, Berkeley, and a graduate of the University of California, Berkeley. I am a graduate of the University of California, Berkeley, and a graduate of the University of California, Berkeley. I am a graduate of the University of California, Berkeley, and a graduate of the University of California, Berkeley. I am a graduate of the University of California, Berkeley, and a graduate of the University of California, Berkeley. I am a graduate of the University of California, Berkeley, and a graduate of the University of California, Berkeley. I am a graduate of the University of California, Berkeley, and a graduate of the University of California, Berkeley. I am a graduate of the University of California, Berkeley, and a graduate of the University of California, Berkeley"}],"usage":{"prompt_tokens":0,"completion_tokens":0,"total_tokens":0}}
```

**Logs**
KO logs from v2.8.0:
```
4:47PM INF Loading model 'facebook/opt-125m' with backend vllm
4:47PM DBG Loading model in memory from file: /build/models/facebook/opt-125m
4:47PM DBG Loading Model facebook/opt-125m with gRPC (file: /build/models/facebook/opt-125m) (backend: vllm): {backendString:vllm model:facebook/opt-125m threads:12 assetDir:/tmp/localai/backend_data context:{emptyCtx:{}} gRPCOptions:0xc0000f6600 externalBackends:map[autogptq:/build/backend/python/autogptq/run.sh bark:/build/backend/python/bark/run.sh coqui:/build/backend/python/coqui/run.sh diffusers:/build/backend/python/diffusers/run.sh exllama:/build/backend/python/exllama/run.sh exllama2:/build/backend/python/exllama2/run.sh huggingface-embeddings:/build/backend/python/sentencetransformers/run.sh mamba:/build/backend/python/mamba/run.sh petals:/build/backend/python/petals/run.sh sentencetransformers:/build/backend/python/sentencetransformers/run.sh transformers:/build/backend/python/transformers/run.sh transformers-musicgen:/build/backend/python/transformers-musicgen/run.sh vall-e-x:/build/backend/python/vall-e-x/run.sh vllm:/build/backend/python/vllm/run.sh] grpcAttempts:20 grpcAttemptsDelay:2 singleActiveBackend:false parallelRequests:false}
4:47PM DBG Loading external backend: /build/backend/python/vllm/run.sh
4:47PM DBG Loading GRPC Process: /build/backend/python/vllm/run.sh
4:47PM DBG GRPC Service for facebook/opt-125m will be running at: '127.0.0.1:38441'
4:47PM DBG GRPC Service state dir: /tmp/go-processmanager400697904
4:47PM DBG GRPC Service Started
4:47PM DBG GRPC(facebook/opt-125m-127.0.0.1:38441): stderr Traceback (most recent call last):
4:47PM DBG GRPC(facebook/opt-125m-127.0.0.1:38441): stderr   File "/build/backend/python/vllm/backend_vllm.py", line 13, in <module>
4:47PM DBG GRPC(facebook/opt-125m-127.0.0.1:38441): stderr     from vllm import LLM, SamplingParams
4:47PM DBG GRPC(facebook/opt-125m-127.0.0.1:38441): stderr   File "/opt/conda/envs/transformers/lib/python3.11/site-packages/vllm/__init__.py", line 3, in <module>
4:47PM DBG GRPC(facebook/opt-125m-127.0.0.1:38441): stderr     from vllm.engine.arg_utils import AsyncEngineArgs, EngineArgs
4:47PM DBG GRPC(facebook/opt-125m-127.0.0.1:38441): stderr   File "/opt/conda/envs/transformers/lib/python3.11/site-packages/vllm/engine/arg_utils.py", line 6, in <module>
4:47PM DBG GRPC(facebook/opt-125m-127.0.0.1:38441): stderr     from vllm.config import (CacheConfig, ModelConfig, ParallelConfig,
4:47PM DBG GRPC(facebook/opt-125m-127.0.0.1:38441): stderr   File "/opt/conda/envs/transformers/lib/python3.11/site-packages/vllm/config.py", line 9, in <module>
4:47PM DBG GRPC(facebook/opt-125m-127.0.0.1:38441): stderr     from vllm.utils import get_cpu_memory, is_hip
4:47PM DBG GRPC(facebook/opt-125m-127.0.0.1:38441): stderr   File "/opt/conda/envs/transformers/lib/python3.11/site-packages/vllm/utils.py", line 11, in <module>
4:47PM DBG GRPC(facebook/opt-125m-127.0.0.1:38441): stderr     from vllm._C import cuda_utils
4:47PM DBG GRPC(facebook/opt-125m-127.0.0.1:38441): stderr ImportError: /opt/conda/envs/transformers/lib/python3.11/site-packages/vllm/_C.cpython-311-x86_64-linux-gnu.so: undefined symbol: _ZN2at4_ops15to_dtype_layout4callERKNS_6TensorEN3c108optionalINS5_10ScalarTypeEEENS6_INS5_6LayoutEEENS6_INS5_6DeviceEEENS6_IbEEbbNS6_INS5_12MemoryFormatEEE
4:48PM ERR Failed starting/connecting to the gRPC service: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp 127.0.0.1:38441: connect: connection refused"
[192.168.96.1]:48316 500 - POST /v1/completions
```

OK logs from v2.7.0:
```
4:51PM INF Loading model 'facebook/opt-125m' with backend vllm
4:51PM DBG Loading model in memory from file: /build/models/facebook/opt-125m
4:51PM DBG Loading Model facebook/opt-125m with gRPC (file: /build/models/facebook/opt-125m) (backend: vllm): {backendString:vllm model:facebook/opt-125m threads:12 assetDir:/tmp/localai/backend_data context:{emptyCtx:{}} gRPCOptions:0xc0001f8800 externalBackends:map[autogptq:/build/backend/python/autogptq/run.sh bark:/build/backend/python/bark/run.sh coqui:/build/backend/python/coqui/run.sh diffusers:/build/backend/python/diffusers/run.sh exllama:/build/backend/python/exllama/run.sh exllama2:/build/backend/python/exllama2/run.sh huggingface-embeddings:/build/backend/python/sentencetransformers/run.sh mamba:/build/backend/python/mamba/run.sh petals:/build/backend/python/petals/run.sh sentencetransformers:/build/backend/python/sentencetransformers/run.sh transformers:/build/backend/python/transformers/run.sh transformers-musicgen:/build/backend/python/transformers-musicgen/run.sh vall-e-x:/build/backend/python/vall-e-x/run.sh vllm:/build/backend/python/vllm/run.sh] grpcAttempts:20 grpcAttemptsDelay:2 singleActiveBackend:false parallelRequests:false}
4:51PM DBG Loading external backend: /build/backend/python/vllm/run.sh
4:51PM DBG Loading GRPC Process: /build/backend/python/vllm/run.sh
4:51PM DBG GRPC Service for facebook/opt-125m will be running at: '127.0.0.1:46551'
4:51PM DBG GRPC Service state dir: /tmp/go-processmanager1567678078
4:51PM DBG GRPC Service Started
4:51PM DBG GRPC(facebook/opt-125m-127.0.0.1:46551): stderr Server started. Listening on: 127.0.0.1:46551
4:51PM DBG GRPC Service Ready
4:51PM DBG GRPC: Loading model with options: {state:{NoUnkeyedLiterals:{} DoNotCompare:[] DoNotCopy:[] atomicMessageInfo:<nil>} sizeCache:0 unknownFields:[] Model:facebook/opt-125m ContextSize:0 Seed:0 NBatch:512 F16Memory:false MLock:false MMap:false VocabOnly:false LowVRAM:false Embeddings:false NUMA:false NGPULayers:0 MainGPU: TensorSplit: Threads:12 LibrarySearchPath: RopeFreqBase:0 RopeFreqScale:0 RMSNormEps:0 NGQA:0 ModelFile:/build/models/facebook/opt-125m Device: UseTriton:false ModelBaseName: UseFastTokenizer:false PipelineType: SchedulerType: CUDA:false CFGScale:0 IMG2IMG:false CLIPModel: CLIPSubfolder: CLIPSkip:0 ControlNet: Tokenizer: LoraBase: LoraAdapter: LoraScale:0 NoMulMatQ:false DraftModel: AudioPath: Quantization: MMProj: RopeScaling: YarnExtFactor:0 YarnAttnFactor:0 YarnBetaFast:0 YarnBetaSlow:0 Type:}
config.json: 100%|██████████| 651/651 [00:00<00:00, 6.97MB/s]
4:51PM DBG GRPC(facebook/opt-125m-127.0.0.1:46551): stdout INFO 02-14 16:51:52 llm_engine.py:70] Initializing an LLM engine with config: model='facebook/opt-125m', tokenizer='facebook/opt-125m', tokenizer_mode=auto, revision=None, tokenizer_revision=None, trust_remote_code=False, dtype=torch.float16, max_seq_len=2048, download_dir=None, load_format=auto, tensor_parallel_size=1, quantization=None, enforce_eager=False, seed=0)
tokenizer_config.json: 100%|██████████| 685/685 [00:00<00:00, 6.97MB/s]
vocab.json: 100%|██████████| 899k/899k [00:00<00:00, 1.74MB/s]
merges.txt: 100%|██████████| 456k/456k [00:00<00:00, 34.5MB/s]
special_tokens_map.json: 100%|██████████| 441/441 [00:00<00:00, 4.48MB/s]
pytorch_model.bin: 100%|██████████| 251M/251M [00:00<00:00, 315MB/s]
4:52PM DBG GRPC(facebook/opt-125m-127.0.0.1:46551): stdout INFO 02-14 16:52:06 llm_engine.py:275] # GPU blocks: 148228, # CPU blocks: 7281
4:52PM DBG GRPC(facebook/opt-125m-127.0.0.1:46551): stdout INFO 02-14 16:52:07 model_runner.py:501] Capturing the model for CUDA graphs. This may lead to unexpected consequences if the model is not static. To run the model in eager mode, set 'enforce_eager=True' or use '--enforce-eager' in the CLI.
4:52PM DBG GRPC(facebook/opt-125m-127.0.0.1:46551): stdout INFO 02-14 16:52:07 model_runner.py:505] CUDA graphs can take additional 1~3 GiB memory per GPU. If you are running out of memory, consider decreasing `gpu_memory_utilization` or enforcing eager mode.
4:52PM DBG GRPC(facebook/opt-125m-127.0.0.1:46551): stdout INFO 02-14 16:52:09 model_runner.py:547] Graph capturing finished in 2 secs.
Processed prompts: 100%|██████████| 1/1 [00:00<00:00,  2.34it/s]
4:52PM DBG Response: {"created":1707929418,"object":"text_completion","id":"273ff23c-6ce0-4fc6-8d89-6bb62e499f73","model":"vllm","choices":[{"index":0,"finish_reason":"stop","text":" J.C. and I am a student at the University of California, Berkeley. I am a graduate of the University of California, Berkeley, and a graduate of the University of California, Berkeley. I am a graduate of the University of California, Berkeley, and a graduate of the University of California, Berkeley. I am a graduate of the University of California, Berkeley, and a graduate of the University of California, Berkeley. I am a graduate of the University of California, Berkeley, and a graduate of the University of California, Berkeley. I am a graduate of the University of California, Berkeley, and a graduate of the University of California, Berkeley. I am a graduate of the University of California, Berkeley, and a graduate of the University of California, Berkeley. I am a graduate of the University of California, Berkeley, and a graduate of the University of California, Berkeley. I am a graduate of the University of California, Berkeley, and a graduate of the University of California, Berkeley"}],"usage":{"prompt_tokens":0,"completion_tokens":0,"total_tokens":0}}
[192.168.96.1]:35012 200 - POST /v1/completions
```

**Comment by mudler** [comment]
waiting on feedback (potential fix is available in master)

**Comment by golgeek** [comment]
Fixed with #1711; thanks!
